### PR TITLE
home-work-2-robert

### PR DESCRIPTION
**Pull request by OK-Ro**

### CODE REVIEW

File: 1-JavaScript/Week2/homework/ex6-totalCost.js [hunk 1]
```diff
@@ -21,22 +21,39 @@ instead!
...
 
-function calculateTotalPrice(/* TODO parameter(s) go here */) {
+function calculateTotalPrice(myCart) {
   // TODO replace this comment with your code
+ 
+  let total = 0;
+  for (let item in myCart) {
+  total += myCart[item];
+ }
+ return `Total: ${total}`;
```
Comment (stasel) [inline 1-JavaScript/Week2/homework/ex6-totalCost.js:41]: Very good and nice logic.
 Try to keep the formatting a little more tidy. Ex: The statement inside the for loop should be more indented to look inside

File: 1-JavaScript/Week2/homework/ex6-totalCost.js [hunk 2]
```diff
@@ -21,22 +21,39 @@ instead!
...
 }
+const totalPrice = calculateTotalPrice(cartForParty);
+console.log(totalPrice);
 
 // ! Test functions (plain vanilla JavaScript)
 function test1() {
   console.log('\nTest 1: calculateTotalPrice should take one parameter');
   // TODO replace this comment with your code
+
```
Comment (stasel) [inline 1-JavaScript/Week2/homework/ex6-totalCost.js:50]: Function looks good but you forgot to write tests. Can you write some tests and try to use Jest to verify that your function works well?

File: 1-JavaScript/Week2/homework/ex7-mindPrivacy.js [hunk 3]
```diff
@@ -30,9 +32,11 @@ const employeeRecords = [
 ];
 
 // ! Function under test
-function filterPrivateData(/* TODO parameter(s) go here */) {
+function filterPrivateData(filteredRecords) {
   // TODO complete this function
+  return filteredRecords.map(({ name, occupation, email }) => ({ name, occupation, email }));
 }
+console.log(filterPrivateData(employeeRecords));
```
Comment (stasel) [inline 1-JavaScript/Week2/homework/ex7-mindPrivacy.js:39]: This `console.log` will never be executed because it's after the `return` statement. Maybe you forgot it to remove it by accident? Not a problem

File: 1-JavaScript/Week2/homework/ex7-mindPrivacy.js [hunk 4]
```diff
@@ -30,9 +32,11 @@ const employeeRecords = [
 ];
 
 // ! Function under test
-function filterPrivateData(/* TODO parameter(s) go here */) {
+function filterPrivateData(filteredRecords) {
   // TODO complete this function
+  return filteredRecords.map(({ name, occupation, email }) => ({ name, occupation, email }));
```
Comment (stasel) [inline 1-JavaScript/Week2/homework/ex7-mindPrivacy.js:37]: Aha! a `map`! That's week 3 material here. Nice job.

**Review by stasel — COMMENTED** [review]
Dear Robert,
After all your hard work I can really say that it paid off. Your homework is excellent! I really see that you understand how to write functions, use string templates and loop on an array. You code it readable and I really like that you add a lot of comments to make sure we all understand what it is doing

HW is almost done, please address my comments in ex4 and ex6 and resubmit again so I can approve it.

👍

File: 1-JavaScript/Week2/homework/ex2-dogYears.js [hunk 5]
```diff
@@ -16,8 +16,12 @@ calculate it!
    ages.
 -----------------------------------------------------------------------------*/
 
-function calculateDogAge(/* TODO parameter(s) go here */) {
+function calculateDogAge(age) {
   // TODO complete this function
+  // dogs age is human years multiplied by 7 years age of dog
+  let dogYears = age * 7;
+  // returning dog age
+  return `your doggie is ${dogYears} years old in dog years!`;
```
[the remark this answers is not in the document]
Comment (OK-Ro) [reply 1-JavaScript/Week2/homework/ex2-dogYears.js:24]: Owe that even makes the code more short

**Review by OK-Ro — COMMENTED** [review]
thanks you very much and i will go through the same homework just get for famillier with all the comments  try to implement them i think will help improve on my code.

**Review by OK-Ro — COMMENTED** [review]
i have done corrections but only the one for testing code that i still have problems need to work on it i still dont know the concept of testing code

File: 1-JavaScript/Week2/homework/ex6-totalCost.js [hunk 6]
```diff
@@ -21,23 +22,36 @@ instead!
...
 function test1() {
   console.log('\nTest 1: calculateTotalPrice should take one parameter');
-  // TODO replace this comment with your code
 }
 
 function test2() {
   console.log('\nTest 2: return correct output when passed cartForParty');
   // TODO replace this comment with your code
 }
+// PLEASE I STILL DONT KNOW THE CONCEPTS OF TESTING CODE HAVE TRIED BUT CANT FIGURE OUT
```
Comment (stasel) [inline 1-JavaScript/Week2/homework/ex6-totalCost.js:54]: Hi Bob, I showed in our Q&A an example to write tests without Jest. Try to achieve that concept. Look at the tests in other exercises in week 2 homework to see more examples.

**Review by stasel — COMMENTED** [review]
Hi Bob, good work implementing the changes, can you please try to write some tests in exercise 6 and also you forgot to handle the edge case in ex4 when calling the function with no parameters

**Comment by OK-Ro** [comment]
homework

**Review by stasel — APPROVED** [review]
Looks good. Thank you for spending the time correcting your homework

✅
